src/components/Review/DecisionPreview/ReviewPreviewModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react'
+import { Modal, Message, Button, Loader, ModalProps } from 'semantic-ui-react'
+import { useLanguageProvider } from '../../../contexts/Localisation'
+import { postRequest } from '../../../utils/helpers/fetchMethods'
+import config from '../../../config'
+import { getItemDisplayComponent, ActionResultPreviewData } from './PreviewItems'
+
+interface PreviewProps extends ModalProps {
+  open: boolean
+  setOpen: (open: boolean) => void
+  decision: string
+  reviewId: number
+  previewData: { [key: string]: any }
+}
+
+const previewEndpoint = `${config.serverREST}/preview-actions`
+
+const ReviewPreviewModal: React.FC<PreviewProps> = ({
+  open,
+  setOpen,
+  decision,
+  reviewId,
+  previewData,
+}) => {
+  const { strings } = useLanguageProvider()
+  const [data, setData] = useState<ActionResultPreviewData[] | null>(null)
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(false)
+
+  useEffect(() => {
+    if (!open) {
+      setData(null)
+      setError(false)
+      return
+    }
+    setLoading(true)
+    fetchPreviews(reviewId, previewData).then((result) => {
+      if (result.error) setError(result.error)
+      else setData(result.displayData)
+      setLoading(false)
+    })
+  }, [open])
+
+  return (
+    <Modal id="preview-modal" open={open} closeOnDimmerClick={false}>
```

Comment:
`closeOnDimmerClick={false}` -- thought I'd make it a bit harder to accidentally close the modal, cos it generates new previews every time it's opened.

Reply:
Hmm, can you add a comment about it in code please

Reply:
👍🏼  f66b942b